robocop/utils/disablers.py

```diff
@@ -0,0 +1,105 @@
+import re
+from collections import defaultdict
+
+
+class DisablersInFile:
+    def __init__(self):
+        self.lastblock = -1
+        self.lines = set()
+        self.blocks = list()
+
+
+class DisablersFinder:
+    def __init__(self, source, linter):
+        self.linter = linter
+        self.file_disabled = False
+        self.any_disabler = False
+        self.disabler_pattern = re.compile(r'robocop: (?P<disabler>disable|enable)=?(?P<rules>[\w\-,]*)')
+        self.rules = defaultdict(lambda: DisablersInFile())  # TODO: implement copy for better perfomance (test it)
+        self._parse_file(source)
+
+    def is_msg_disabled(self, msg):
+        if not self.any_disabler:
```

Comment:
this tree of conditions should save some computing time, but I need to test it later on larger test data base